pkg/operator/controller/certificate/ca.go

```diff
@@ -41,6 +43,13 @@ func (r *reconciler) ensureRouterCASecret() (*corev1.Secret, error) {
 		r.recorder.Event(new, "Normal", "CreatedWildcardCACert", "Created a default wildcard CA certificate")
 		return new, nil
 
+	} else if !reflect.DeepEqual(current.Annotations, desired.Annotations) {
+		updated := current.DeepCopy()
+		updated.Data = desired.Data
+		updated.Annotations = desired.Annotations
```

Comment:
You may need to insert/merge the annotations rather than overwriting all of them.  Adding a unit test for `addition of a new annotation to existing annotations` would be helpful.

Reply:
I'd expect `desired` annotations to be an full list of annotation, not just the required minimum. Is there an example of optional annotation we want to keep in these secrets?